shoes-core/lib/shoes/line.rb

```diff
@@ -19,6 +21,26 @@ def create_dimensions(point_a, point_b)
       style[:y2] = point_b.y
     end
 
+    # Check out http://math.stackexchange.com/questions/60070/checking-whether-a-point-lies-on-a-wide-line-segment
+    # for explanations how the algorithm works
+    def in_bounds?(x, y)
+      # c is (x, y)
+      left_most, right_most = point_a.x < point_b.x ? [point_a, point_b] : [point_b, point_a]
+      vec_left_c = Vector[left_most.x - x, left_most.y - y]
```

Comment:
I mean, maybe something like
```
class Point
  ...
  def -(other)
    other = other.to_a
    Point.new(x - other[0], y - other[1])
  end
  
  def to_a
    [x, y]
  end
end

```
And then it would allow to
```
p0 - p1
p0 - [1, 2]
```